### ISSUE TITLE
question (Reddit) : How to patch Reddit v2024.30.0 and above?

### ISSUE THREAD

**Issue by Aryan-Chauhan** [inotia00/ReVanced_Extended#2289]
### Application

Other

### Question description

On APKMirror, Reddit v2024.30.0 and above are only available as split APKs.

v2024.29.0 is the last version that has a normal APK.

Is there any way to patch v2024.30.0 and above?

Thanks!

### Acknowledgements

- [X] This issue is not a duplicate of an existing question.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
You can apply the patch after converting it to universal apk using anti-split

I plan to support this in the next rvx-builder release https://github.com/inotia00/ReVanced_Extended/issues/2292

**Comment by Aryan-Chauhan** [comment]
Thanks!

For now, I used the below app and successfully patched v2024.32.0

https://github.com/AbdurazaaqMohammed/AntiSplit-M

**Comment by i486** [comment]
> Thanks!
> 
> For now, I used the below app and successfully patched v2024.32.0
> 
> https://github.com/AbdurazaaqMohammed/AntiSplit-M

Did you have to turn on arsclib to get it to work?

I'm using [rvx-builder v3.15.1](https://github.com/inotia00/rvx-builder/releases/tag/v3.15.1) to patch the latest Reddit (v2024.32.0). When arsclib is enabled, everything works fine. 

However, if I disable arsclib, my build gets stuck at the `Signing APK` step:

Interestingly, I can build v2024.29.0 without any issues, even with arsclib disabled.

```
[builder] INFO: Compiling modified resources
[builder] INFO: Aligning APK
[builder] INFO: Signing APK
```
The reason I’m asking is that I saw a comment [here](https://github.com/inotia00/ReVanced_Extended/issues/2246#issuecomment-2254150017) suggesting that arsclib is no longer needed for versions v2024.28.0 and later. I’m uncertain whether the issue with v2024.30+ is due to arsclib still being required, a bug in rvx-builder, or if it might be related to my use of Windows 7 (though I doubt this last possibility).

**TL;DR**

- Using the latest rvx-builder:
  - v2024.30.0+ gets stuck at the `Signing APK` step when arsclib is disabled.
  - v2024.29.0 works fine with arsclib disabled.
  - v2024.XX.XX works correctly with arsclib enabled as expected.

**Comment by inotia00** [comment]
1. If you turn off ARSCLib setting, the default repository (inotia00/revanced-patches) will be used, in which case the recommended version is **Reddit 2024.17.0**.

You tried to patch from a non-recommended version.

2. If you use the public ReVanced source(ReVanced/revanced-patches), you need to remove the keystore.

This is mentioned in the [release notes of rvx-builder](https://github.com/inotia00/rvx-builder/releases/tag/v3.15.1)
